02-peer-protocol.md

```diff
@@ -420,16 +412,91 @@ it is possible for the fee for the exchanged witness data to be underpaid.
 It is the responsibility of the sending peer to correctly account for the
 required fee, e.g. a multisig witness stack whose weight exceeds 110.
 
+### The `tx_init_rbf` Message
+
+This message initiates a replacement of the transaction after it's been
+completed.
+
+1. type: 72 (`tx_init_rbf`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u32`:`locktime`]
+   * [`u32`:`feerate`]
+
+1. `tlv_stream`: `tx_init_rbf_tlvs`
+2. types:
+    1. type: 0 (`shared_output_contribution`)
+    2. data:
+        * [`tu64`:`satoshis`]
+
+#### Requirements
+
+The sender:
+  - MUST set `feerate` greater than or equal to 65/64 times the `feerate`
+    of the previously constructed transaction, rounded down.
+
+The recipient:
+  - MUST respond either by failing the negotiation or with `tx_ack_rbf`.
+  - MUST fail the negotiation if:
+    - the `feerate` is not greater than 65/64 times `feerate` of the last
+      successfully constructed transaction
+  - MAY fail the negotiation for any reason
+
+#### Rationale
+
+`feerate` is the feerate this transaction will pay. It must be at least
+1/64 greater than the last used `feerate`, rounded down to the nearest
+satoshi to ensure there is progress.
+
+E.g. if the last `feerate` was 520, the next sent `feerate` must be 528
+(520 * 65 / 64 = 528.125, rounded down to 528).
+
+If the previous transaction confirms in the middle of an RBF attempt,
+the attempt MUST be abandoned.
+
+`shared_output_contribution` is the amount of satoshis that this peer
+will contribute to the shared output of the transaction, when there is
+such an output. Note that it may be different from the contribution
+made in the previously completed transaction.
+
+### The `tx_ack_rbf` Message
+
+1. type: 73 (`tx_ack_rbf`)
+2. data:
+    * [`channel_id`:`channel_id`]
+
+1. `tlv_stream`: `tx_ack_rbf_tlvs`
+2. types:
+    1. type: 0 (`shared_output_contribution`)
+    2. data:
+        * [`tu64`:`satoshis`]
+
+#### Requirements
+
+The recipient:
+  - MUST either fail the negotiation or transmit a `tx_add_input` message,
+    restarting the interactive tx collaboration protocol.
+
+#### Rationale
+
+`shared_output_contribution` is the amount of satoshis that this peer
+will contribute to the shared output of the transaction, when there is
+such an output. Note that it may be different from the contribution
+made in the previously completed transaction.
+
+It's recommended that a peer, rather than fail the RBF negotiation due to
+a large feerate change, instead sets their `shared_output_contribution` to
+zero, and decline to participate further in the transaction.
```

Comment:
ACK. This is reasonable.

It has a small interaction with liquidity ad proposal; we'd want to add a check to the liquidity ad negotiation where a change in the `shared_output_contribution` is less than the negotiated liquidity lease, you'd fail the RBF.